src/services/UserInfoRequestProcessor.ts

```diff
@@ -0,0 +1,195 @@
+		}
+
+		// add sessionId to all subsequent log messages
+		this.logger.appendKeys({ sessionId });
+		
+		const session: ISessionItem | undefined = await this.BavService.getSessionById(sessionId);
+		if (!session) {
+			this.logger.error("No session found", {
+				messageCode: MessageCodes.SESSION_NOT_FOUND,
+			});
+			return new Response(HttpCodesEnum.UNAUTHORIZED, "Unauthorized");
+		}
+
+		// add govuk_signin_journey_id to all subsequent log messages
+		this.logger.appendKeys({ govuk_signin_journey_id: session.clientSessionId });
+
+		this.logger.info("Found Session:", {
+			// note: we only log specific non-PII attributes from the session object:
+			session: {
+				authSessionState: session.authSessionState,
+				accessTokenExpiryDate: session.accessTokenExpiryDate,
+				attemptCount: session.attemptCount,
+				authorizationCodeExpiryDate: session.authorizationCodeExpiryDate,
+				createdDate: session.createdDate,
+				expiryDate: session.expiryDate,
+				redirectUri: session.redirectUri,
+			},
+		});
+  
+		this.metrics.addMetric("found session data", MetricUnits.Count, 1);
+
+		const personInfo: PersonIdentityItem | undefined = await this.BavService.getPersonIdentityBySessionId(sessionId, this.personIdentityTableName);
+		if (!personInfo) {
+			this.logger.error("No person found with this session ID", {
+				messageCode: MessageCodes.PERSON_NOT_FOUND,
+			});
+			return new Response(HttpCodesEnum.UNAUTHORIZED, "Unauthorized");
+		}
+
+		this.logger.info("Found person Identity data");
+  
+		this.metrics.addMetric("found person identity data", MetricUnits.Count, 1);
+
+		// Validate the AuthSessionState to be "BAV_ACCESS_TOKEN_ISSUED"
+		if (session.authSessionState !== AuthSessionState.BAV_ACCESS_TOKEN_ISSUED) {
+			this.logger.error("Session is in wrong Auth state", {
+				// note: we only log specific non-PII attributes from the session object:
+				expectedSessionState: AuthSessionState.BAV_ACCESS_TOKEN_ISSUED,
+				session: {
+					authSessionState: session.authSessionState,
+				},
+				messageCode: MessageCodes.INCORRECT_SESSION_STATE,
+			});
+			return new Response(HttpCodesEnum.UNAUTHORIZED, "Unauthorized");
+		}
+
+		// Person info required for VC
+		const names = personInfo.name[0].nameParts;
+
+		// Validate the User Info data presence required to generate the VC
+		if (names && names.length > 0 && personInfo.sortCode && personInfo.accountNumber) {
+
+			//Generate VC and create a signedVC as response back to IPV Core.
+			const signedJWT = await this.verifiableCredentialService.generateSignedVerifiableCredentialJwt(session, names, {
+				sortCode: personInfo.sortCode,
+				accountNumber: personInfo.accountNumber,
+			}, absoluteTimeNow);
+
+			// Add metric and send TXMA event to the sqsqueue
+			this.metrics.addMetric("Generated signed verifiable credential jwt", MetricUnits.Count, 1);
+
+			this.BavService.updateSessionAuthState(session.sessionId, AuthSessionState.BAV_CRI_VC_ISSUED)
```

Comment:
There is a linting error about this not being awaited

Reply:
Done